router/transformer/transformer.go

```diff
@@ -322,6 +325,223 @@ func (trans *HandleT) makeHTTPRequest(ctx context.Context, url string, payload [
+		req, err := http.NewRequestWithContext(ctx, requestMethod, postInfo.URL, payload)
+		if err != nil {
+			errStr := fmt.Sprintf(`400 Unable to construct "%s" request for URL : "%s"`, requestMethod, postInfo.URL)
+			trans.logger.Error(errStr)
+			return nil, fmt.Errorf(errStr)
+			// return &utils.SendPostResponse{
+			// 	StatusCode:   400,
+			// 	ResponseBody: []byte(fmt.Sprintf(`400 Unable to construct "%s" request for URL : "%s"`, requestMethod, postInfo.URL)),
+			// }
+		}
+
+		// add queryparams to the url
+		// support of array type in params is handled if the
+		// response from transformers are "," seperated
+		queryParams := req.URL.Query()
+		for key, val := range requestQueryParams {
+
+			// list := strings.Split(valString, ",")
+			// for _, listItem := range list {
+			// 	queryParams.Add(key, fmt.Sprint(listItem))
+			// }
+			formattedVal := handleQueryParam(val)
+			queryParams.Add(key, formattedVal)
+		}
+
+		req.URL.RawQuery = queryParams.Encode()
+		headerKV := postInfo.Headers
+		for key, val := range headerKV {
+			req.Header.Add(key, val.(string))
+		}
+
+		req.Header.Add("User-Agent", "RudderLabs")
+
+		// This is being done to facilitate compatible comparison
+		// As map[string][]string is the data-type for headers in golang
+		// But headers is an object in Javascript, hence we need to level the plane for effective comparison
+		headersMap := make(map[string]string)
+		for k, v := range req.Header {
+			headersMap[strings.ToLower(k)] = string(v[0])
+		}
+
+		rtDelPayload := RouterDelPayload{
+			Data:     string(payloadBytes),
+			Method:   requestMethod,
+			Params:   req.URL.Query(),
+			Endpoint: postInfo.URL,
+			Headers:  headersMap,
+		}
+
+		return rtDelPayload, nil
+	}
+	return nil, fmt.Errorf("this type of request currently is not supported")
+}
+
+//temp solution for handling complex query params
+func handleQueryParam(param interface{}) string {
+	switch p := param.(type) {
+	case string:
+		return p
+	case map[string]interface{}:
+		temp, err := jsonfast.Marshal(p)
+		if err != nil {
+			return fmt.Sprint(p)
+		}
+
+		jsonParam := string(temp)
+		return jsonParam
+	default:
+		return fmt.Sprint(param)
+	}
+}
+
+type RouterDelPayload struct {
+	// The data of the router payload after some changes are applied while preparing the event delivery request
+	Data interface{} `json:"data"`
+	// The event delivery request method
+	Method string `json:"method"`
+	// The query parameters for the event delivery
+	Params interface{} `json:"params"`
+	// The url to which the event has to be sent
+	Endpoint string `json:"endpoint"`
+	// The headers for the event delivery request
+	Headers interface{} `json:"headers"`
+}
+
+type ProxyTestReqBody struct {
+	// The event delivery request payload formed by Router
+	RouterDeliveryPayload RouterDelPayload
+	// The payload which will be used by transformer to form event delivery request payload
+	ProxyRequestPayload integrations.PostParametersT
```

Comment:
Can we name them a bit better? I've suggested the names I'm expecting in the transformer but feel free to change them.
```suggestion
	// The event delivery request payload formed by Router
	destinationRequestPayload RouterDelPayload
	// The payload which will be used by the transformer to form the event delivery request payload
	deliveryPayload integrations.PostParametersT
```

Reply:
We can add json definitions for them like how they are in `RouterDelPayload`
Do you think they have to be interchanged ?
`RouterDeliveryPayload` -> ``json:"deliveryPayload"``
`ProxyRequestPayload` -> ``json:"destinationRequestPayload"``

Reply:
I named them as per the below notion:

1. Request payload send to the destination - `destinationRequestPayload`
2. Input data which router operates on - `deliveryPayload`. This sounds a bit generic, maybe we can name it something like `transformerProcResponse`?

The combined DS can be called `ProxyTestRequestPayload`.

Reply:
Done